### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by regisss**
# What does this PR do?

This PR updates the documentation:

- Explanations about how to do pretraining are added.
- A working DeepSpeed configuration is given in the README of each example.
- Swin is added to the table of validated models.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: docs/source/pretraining.mdx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+
+# Pretraining a model with Optimum Habana
+
+Pretraining a model can be done as easily as fine-tuning it.
+
+For fine-tuning, you instantiate the model as follows:
```
Comment (philschmid) [inline docs/source/pretraining.mdx:22]: Is that needed? We could just add the "pre-training" snippet

**Review by philschmid — COMMENTED** [review]
Left some comments. Nice idea with adding a Pretraining section!

**Comment by regisss** [comment]
@philschmid I added a pretraining example using your preprocessed dataset (so that the code snippet is not too long)

**Review by libinta — APPROVED** [review]

File: docs/source/pretraining.mdx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+Pretraining a model from Transformers, like BERT, is as easy as fine-tuning it.
+The model should be instantiated from a configuration with `.from_config` and not from a pretrained checkpoint with `.from_pretrained`:
+```python
+from transformers import AutoConfig, AutoModelForXXX
+
+config = AutoConfig.from_pretrained(model_name)
+model = AutoModelForXXX.from_config(config)
+```
+with XXX the task to perform, such as `ImageClassification` for example.
```
Comment (philschmid) [inline docs/source/pretraining.mdx:28]: I would make is a concrete example ,e.g. for `bert-base-uncased` or `gpt2`

**Review by philschmid — APPROVED** [review]
LGTM

**Pull request merged**